src/oldToNewHash.ts

```diff
@@ -0,0 +1,330 @@
+import { ipfs } from "@graphprotocol/graph-ts"
+
+export function getNewHash(ipfsHash: string): string {
+
+    if (ipfsHash == "QmSWxFAHpjRiydEBZdeQfzWmpLuc3YHBTnvYSkxZyM6Bj5")
+        ipfsHash = "QmZJnjgbH8z5QbLzNcPaCK4bMzvye5MpPTS4vSpB55XBfm"
```

Comment:
Why would you do this?! This is abysmal. This is data for God's sake, you don't need all these `if`s.
`ipfsHashMap.json`
```json
{
  "oldHashA": "newHashA",
  "oldHashB": "newHashB",
  "oldHashC": "newHashC",
  ...
}
```
`src/oldToNewHash.ts`
```js
import hashMap from "./ipfsHashMap.json"

export function getNewHash(ipfsHash: string): string {
  if (ipfsHash in hashMap) {
    return hashMap[ipfsHash]
  }
  return ipfsHash
}
```
And add a comment explaining why this function is needed in the first place. Something like:
```js
/* This method is required due to a bug in uploading Dapp metadata to IPFS.
 * Old content hashes are retained and cannot be changed due to how contract was written.
 * This maps old invalid hashes to new valid contents hashes for IPFS.
 * For more details see: https://github.com/status-im/infra-ipfs/issues/4 */
```

Reply:
i wanted to make sure it works s i just wrote a python script to generate these "ifs". This hashMap way looks amazing but i am getting error while building now.
```bash
⠋ Compile subgraphERROR TS1141: String literal expected.

 import hashMap from "./ipfsHashMap.json";
 ~~~~~~
 in src/oldToNewHash.ts(1,0)

✖ Failed to compile subgraph: Failed to compile data source mapping: Parse error

```

Reply:
Node.js does JSON imports by default. I guess TypeScripts is a bit retarded:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#new---resolvejsonmodule
Just add to `tsconfig.json`.

Reply:
In general if something can be data instead of code it should be data.
Also, I showed you how to make the JSON file I suggested using [jq](https://stedolan.github.io/jq/) in https://github.com/dap-ps/discover/issues/42:
``` 
 > curl -s https://dap.ps/metadata/all \
    | jq '[.[] | { key: .hash, value: .ipfsHash } | select(.key != .value)] | from_entries' \
    | head
{                                                                                                                                            
  "QmSWxFAHpjRiydEBZdeQfzWmpLuc3YHBTnvYSkxZyM6Bj5": "QmZJnjgbH8z5QbLzNcPaCK4bMzvye5MpPTS4vSpB55XBfm",
  "QmNgNAvHccH7jAbvpCZFLKGNJ8pCZm7Jepmibg3tCSCBPC": "QmVoryjXtawJVRjELzRfmpXgGkzR6U2RJvXLjMXnDWUQrT",
  "QmRFdYmKphoKHh1TocKdqDiY8D1RomRx3mS55fhwY1jM2H": "QmRCXSJLNfJpWzC7hZBQoLMCdbzuSrs8PNR3sJsKREiRSV",
  "QmWXKBHcWqwyTM7hxWXSqVcTV2FVdb6NstsuUb9UVSjrWZ": "QmQFMrVpHweydbCYXwHq835egvjyTJDGJwMjgsopCJaHvz",
  "QmVmY5K5KffCCzNmMsZp7iDefNPjtQomh4eGERso7ShSFk": "Qme7uzZ7V66n93hKJWWaWs3RhH5oHWJqPtsS8DrvGRZZu6",
...(omitted)...
```
But Python is certainly fine for this.
If you're well versed in Vim macros you could also just take the whole output of `/metadata/all` and clean it up using it.

Reply:
`jq` is mindfuck, i never knew i could extract data like that. And yeah i agree with the point that if it can be data then it should be data.

Reply:
but build still failing, i tried this already https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#new---resolvejsonmodule
I changed the import statement to `import * as hashMap from "./ipfsHashMap.json";` so that i don't get parse error but now i am getting error like `✖ Failed to compile subgraph: Failed to compile data source mapping: Import file 'src/ipfsHashMap.json.ts' not found.
`
I think it is happening because graph is actually using assembly script so different compiler than typescript

Reply:
Well, fuck JSON then, just make this file something like:
`src/ipfsHashMap.ts`
```js
const ipfsHashMap = {
  "oldHashA": "newHashA",
  "oldHashB": "newHashB",
  "oldHashC": "newHashC",
  ...
}

export default ipfsHashMap
```
Same thing essentially.

Reply:
Although, if the error is:
```
Failed to compile data source mapping: Import file 'src/ipfsHashMap.json.ts' not found.
```
Then maybe the proper JSON import line is:
```
import hashMap from "./ipfsHashMap"
```
But whatever, just do the TS module with a const object and call it a day. Simpler.

Reply:
did that and now some new error `Error: Failed to compile data source mapping: assertion failed` and it's not even showing which line

Reply:
>did that

What is "that"? Made it into a normal ts file?

Reply:
The way it looks right looks good to me. Does it work?

Reply:
No gives assertion failed build error

Reply:
How are you testing this?

Reply:
If you look at readme, it has setup instructions. Just clone my fork and then try building the graph.
When you do graph build in my project, it gives the error.

Reply:
@jakubgs done!! Working now :fireworks: